### PR TITLE
#1135 Add POST restful api of streamApp property

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1135)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: docs/rest_interface.md [hunk 1]
```diff
@@ -2784,6 +2792,68 @@ Currently, this api is only used for changing jar name.
...
+
+*POST /v0/stream/property*
+
+**Example Request**
+
+1. jarId (**string**) — the used jar id
+1. name (**option(string)**) — new streamApp name ; default is "Untitled stream app"
+1. from (**option(array(string))**) — new source topics ; default is empty
+1. to (**option(array(string))**) — new target topics ; default is empty
+1. instances (**option(int)**) — new number of running streamApp ; default is 0
```
Comment (chia7712) [inline docs/rest_interface.md]: Is **0** a valid value to us? If not, we can't use **0** as a default value. The default value should be a legal stuff.

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/StreamRoute.scala [hunk 2]
```diff
@@ -310,10 +268,10 @@ private[configurator] object StreamRoute {
                           else
                             Future.successful(
                               previous.copy(
-                                name = req.name,
-                                instances = req.instances,
-                                from = req.from,
-                                to = req.to,
+                                name = if (req.name.isDefined) req.name.get else previous.name,
+                                instances = if (req.instances.isDefined) req.instances.get else previous.instances,
```
Comment (chia7712) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/StreamRoute.scala]: getOrElse is more clear

Comment (saivirtue) [reply ohara-configurator/src/main/scala/com/island/ohara/configurator/route/StreamRoute.scala]: :)

File: docs/rest_interface.md (hunk 1 above)
Comment (saivirtue) [reply docs/rest_interface.md]: make sense. edit the default value to 1

**Comment by saivirtue** [comment]
retry

**Pull request merged**